app/services/host_counter.rb

```diff
@@ -28,14 +28,16 @@ def cache
 
   def counted_hosts
     hosts_scope = Host::Managed.reorder('')
+    grouping = "#{@association}_id"
     case @association.to_s
     when 'organization', 'location'
       # If we are on /organizations or /locations, this allows to display the
       # count for hosts not in the current organization & location.
       hosts_scope = hosts_scope.unscoped
```

Comment:
Should we add the table name always?
```suggestion
      hosts_scope = hosts_scope.unscoped
      grouping.prepend("#{Host.table_name}.")
```

Reply:
Perhaps. Fore sure I'm not as familiar as you're with the codebase so feel free to add extra code and/or tests to the MR :)